pkg/api/index.go

```diff
@@ -159,7 +159,15 @@ func (hs *HTTPServer) getNavTree(c *models.ReqContext, hasEditPerm bool) ([]*dto
 		{Text: "Divider", Divider: true, Id: "divider", HideFromTabs: true},
 		{Text: "Manage", Id: "manage-dashboards", Url: setting.AppSubUrl + "/dashboards", Icon: "sitemap"},
 		{Text: "Playlists", Id: "playlists", Url: setting.AppSubUrl + "/playlists", Icon: "presentation-play"},
-		{Text: "Snapshots", Id: "snapshots", Url: setting.AppSubUrl + "/dashboard/snapshots", Icon: "camera"},
+	}
+
+	if hasEditPerm {
```

Comment:
I just tried with viewer role and I can actually create snapshots so don't think this. Sorry, but think snapshots shouldn't have any permission check here. WDYT?

Reply:
When created a snapshot as viewer I can (in master) see the snapshots I've created:
![image](https://user-images.githubusercontent.com/1668778/99380380-27baf200-28ca-11eb-848d-f6b95b59c5e3.png)

Reply:
Thanks for highlighting this Marcus. Turns out I misread the original issue which states the snapshot table is always empty for an unauthenticated user.

Based on this I'm guessing we either change the logic here to remove the snapshots menu item if there isn't a logged in user or revert this backend change and place a note in the frontend stating something like "you need to be logged in to see snapshots". What do you think?

Reply:
Not authenticated - no snapshot menu item should work I think